### ISSUE TITLE
Cannot set WorkDirectory of test from external code because TestExecutionContext.DefaultWorkDirectory is internal

### ISSUE THREAD

**Issue by KrzysFR** [nunit/nunit#2322]
Since the changes in #2072, I'm not able to specify the Working Directory of a test from an external runner anymore.

I have a basic helper library that runs tests directly from any `void Main()`, and it needs to be able to specify the working directory for each test (value passed from the build server). Previously, I could set it on the `TestExecutionContext.WorkDirectory` property which has been removed.

The code now use the static property `TestContext.DefaultWorkDirectory`, but it is `internal` and cannot be changed from external code. The only code that seems to be able to change this value is in `DefaultTestAssemblyBuilder`, but I would prefer being able to setup and run the test methods myself.

Is there some other way to specify the working directory of each test that I missed?

**Comment by CharliePoole** [comment]
The NUnit WorkDirectory is not the working directory (CurrentDirectory) it is merely the place to which NUnit saves any files it creates, like the test result XML for example.

Since you are not running the tests under the NUnit engine's control, it doesn't seem that the WorkDirectory would have any purpose for you.

**Comment by KrzysFR** [comment]
Ok, so if I don't use these features, I can simply ignore this setting completely and let NUnit decide.

My very basic test runner was written for NUnit 2.x and I remember having to change things a bit when 3.0 was released and the API changed. I may have mixed this property with the CurrentDirectory of the test then (I used the same value for both).

I'll check to see if I'm still able to control the CurrentDirectory of the test itself.
